File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/Handlers/CreateOrderCommandHandler.cs
using MediatR;
using MediatorDesignPattern.Example2.Commands;
using MediatorDesignPattern.Example2.Models;
using MediatorDesignPattern.Example2.Repositories;

namespace MediatorDesignPattern.Example2.Handlers;

/// <summary>
/// Handler for CreateOrderCommand
/// IRequestHandler interface tells MediatR that this is a handler
/// </summary>
public class CreateOrderCommandHandler : IRequestHandler<CreateOrderCommand, Order>
{
    private readonly IOrderRepository _orderRepository;

    public CreateOrderCommandHandler(IOrderRepository orderRepository)
    {
        _orderRepository = orderRepository;
    }

    public Task<Order> Handle(CreateOrderCommand request, CancellationToken cancellationToken)
    {
        var order = new Order
        {
            Id = Guid.NewGuid(),
            CustomerName = request.CustomerName,
            ProductName = request.ProductName,
            Quantity = request.Quantity,
            Price = request.Price,
            Status = OrderStatus.Pending,
            CreatedAt = DateTime.UtcNow
        };

        _orderRepository.Add(order);

        Console.WriteLine($"[HANDLER] New order created: {order.Id}");
        
        return Task.FromResult(order);
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/src/Handlers/CreateOrderCommandHandler.ts
import { requestHandler, RequestHandler } from 'mediatr-ts';
import { CreateOrderCommand } from '../Commands/CreateOrderCommand';
import { Order, OrderStatus } from '../Models/Order';
import { InMemoryOrderRepository } from '../Repositories/InMemoryOrderRepository';

const repository = new InMemoryOrderRepository();
export { repository as orderRepository };

@requestHandler(CreateOrderCommand)
export class CreateOrderCommandHandler implements RequestHandler<CreateOrderCommand, Order> {
  async handle(request: CreateOrderCommand): Promise<Order> {
    const order = new Order({
      id: Math.random().toString(36).substring(2, 10).toUpperCase(),
      customerName: request.customerName,
      productName: request.productName,
      quantity: request.quantity,
      price: request.price,
      status: OrderStatus.Pending,
      createdAt: new Date(),
    });

    repository.add(order);
    console.log(`[HANDLER] New order created: ${order.id}`);
    return order;
  }
}
